src/averageAges.js

```diff
@@ -1,61 +1,69 @@
 'use strict';
 
 /**
- * Implement calculateMenAverageAge function
- *
- * Function returns average age of men in array. If `century` is specified then
- * function calculates average age only for men who died in this century
- *
- * To calculate century:
- * Divide year of person's death by 100: Math.ceil(person.died / 100)
  *
  * @param {object[]} people
  * @param {number} century - optional
  *
  * @return {number}
  */
 function calculateMenAverageAge(people, century) {
-  // write code here
-  // learn how to use array methods like .filter .map .some .every .find .reduce
-  // avoid using loop and forEach
-  // replace `if ()` statement with &&, || or ?:
-  // without nesting
+  const menList = century
+    ? people.filter(person => person.sex === 'm'
+      && Math.ceil(person.died / 100) === century)
+    : people.filter(person => person.sex === 'm');
+
+  const averageAge = (menList
+    .map(men => men.died - men.born)
+    .reduce((a, b) => a + b)) / menList.length;
+
+  return averageAge;
```

Comment:
You repeat this 2 times, mb would be better if you move this to the separate function?
<img width="435" alt="image" src="https://user-images.githubusercontent.com/91411692/227918026-d6557453-fce2-4689-aea6-9d1f57af12d9.png">
```suggestion
  const averageAge =  getAverageAge(menList);

  return averageAge;
```